modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCScopeValidationTest.java

```diff
@@ -0,0 +1,251 @@
+        testAuthentication(playgroundApp);
+        Assert.assertEquals(claimsToGetConsent, "0_Email,1_First Name",
+                "Requested claims were not prompted to ask the consent.");
+        Assert.assertNotNull(scopes);
+        Assert.assertEquals(INTERNAL_LOGIN_SCOPE, scopes, "'" + INTERNAL_LOGIN_SCOPE + "' should return " +
+                "with the redirect URL");
+        Assert.assertFalse(scopes.contains(SYSTEM_SCOPE), "'" + SYSTEM_SCOPE + "' scopes should not exist after the validation.");
+        Assert.assertFalse(scopes.contains(INTERNAL_USER_MGT_UPDATE_SCOPE),
+                "'" + INTERNAL_USER_MGT_UPDATE_SCOPE + "' scopes should not exist after the validation. " +
+                        "Since user does not have permission for this scope.");
+        Assert.assertFalse(scopes.contains(INVALID_SCOPE), "'" + INVALID_SCOPE + "' scopes should not exist" +
+                " after the validation. This scope does not exist.");
+
+    }
+
+    @Test(groups = "wso2.is", dependsOnMethods = {"testScopeValidationWithNormalUser"}, description = "Test scope" +
+            " validation with an admin user.")
+    public void testScopeValidationWithAdminUser() throws Exception {
+        testSendAuthenticationRequest(playgroundApp, client);
+        testAuthentication(playgroundApp);
+        Assert.assertEquals(claimsToGetConsent, "0_Email,1_First Name",
+                "Requested claims were not prompted to ask the consent.");
+        Assert.assertNotNull(scopes);
+        String[] scopesArray = scopes.split(" ");
+        Assert.assertTrue(ArrayUtils.contains(scopesArray, INTERNAL_LOGIN_SCOPE), "'" + INTERNAL_LOGIN_SCOPE +
+                "' should return with the redirect URL");
+        Assert.assertTrue(ArrayUtils.contains(scopesArray, INTERNAL_USER_MGT_UPDATE_SCOPE),
+                "'" + INTERNAL_USER_MGT_UPDATE_SCOPE + "' should return with the redirect URL.");
+        Assert.assertFalse(ArrayUtils.contains(scopesArray, SYSTEM_SCOPE), "'" + SYSTEM_SCOPE + "' scopes " +
+                "should not exist after the validation.");
+        Assert.assertFalse(ArrayUtils.contains(scopesArray, INVALID_SCOPE), "'" + INVALID_SCOPE +
+                "' scopes should not exist after the validation. This scope does not exist.");
+    }
+
+    private void initUser(String role) {
+
+        user = new OIDCUser(OIDCUtilTest.username, OIDCUtilTest.password);
+        user.setProfile(OIDCUtilTest.profile);
+        user.addUserClaim(OIDCUtilTest.emailClaimUri, OIDCUtilTest.email);
+        user.addUserClaim(OIDCUtilTest.firstNameClaimUri, OIDCUtilTest.firstName);
+        user.addUserClaim(OIDCUtilTest.lastNameClaimUri, OIDCUtilTest.lastName);
+        user.addRole(role);
+    }
+
+    private OIDCApplication initApplication() {
+
+        playgroundApp = new OIDCApplication(OIDCUtilTest.playgroundAppOneAppName,
+                OIDCUtilTest.playgroundAppOneAppContext,
+                OIDCUtilTest.playgroundAppOneAppCallBackUri);
```

Comment:
The formatting here looks off. Please check

Reply:
Fixed with https://github.com/wso2/product-is/pull/13004/commits/6215fdfab95c193b90112d9a983ea3bd0113ee46